### PR TITLE
Issue #9: Added element and classname configuration options

### PR DESCRIPTION
**Pull request by michaelBenin**
Config properties in constructor:

```
this.markupClassName = config.markupClassName || "___iso-html___";
this.markupElement = config.markupElement || "div";
this.dataClassName = config.dataClassName || "___iso-state___";
this.dataElement = config.dataElement || "div";
```

Also I saw that on render, we create a new Iso, is this intentional? Also added configuration option to the render call as well. 

Examples updated. Manually tested it on date-time. Would like to see more tests, linting, and documentation.

### CODE REVIEW

**Comment by michaelBenin** [comment]
Ah there's an issue with the date-time. Let me dig into this further.

File: src/iso.js [hunk 1]
```diff
@@ -35,8 +39,8 @@ ${data}
     )
   }
 
-  static render(html, state = {}, meta = {}) {
-    return new Iso().add(html, state, meta).render()
+  static render(config = {markupClassName: '___iso-html___', markupElement: 'div', dataClassName: '___iso-state___', dataElement: 'div'}, html, state = {}, meta = {}) {
```
Comment (goatslacker) [inline src/iso.js]: can we move this to the end so its not a breaking change?

File: src/iso.js [hunk 2]
```diff
@@ -4,7 +4,11 @@ const each = (x, f) => Array.prototype.forEach.call(x, f)
 const parse = (node, x) => JSON.parse(node.getAttribute(x))
 
 export default class Iso {
-  constructor() {
+  constructor(config) {
+    this.markupClassName = config.markupClassName || '___iso-html___'
```
Comment (goatslacker) [inline src/iso.js]: i like this :)

**Comment by michaelBenin** [comment]
Ok I figured it out, the context of this was broken in the class instance. 

Specifically: https://github.com/goatslacker/iso/pull/15/files#diff-338ebbd84df23821b105676cbab7154bR51

I'm just now moving into using ES6/ES2015/Babel. 

Fixed it with:

```
  static bootstrap(onNode, markupClassName='___iso-html___', dataClassName='___iso-state___') {
```

Is there a better way?

Added those arguements at the end to make it backwards compatible. 

Will update the render method as well. 

Fix should be in later tonight around 8 or 9 EST.

**Comment by michaelBenin** [comment]
I'll reopen the pr and squash the commits when it's all cleaned up.

File: src/iso.js (hunk 1 above)
Comment (michaelBenin) [inline src/iso.js]: :+1:

File: src/iso.js [hunk 3]
```diff
@@ -35,17 +39,18 @@ ${data}
...
   }
 
-  static render(html, state = {}, meta = {}) {
-    return new Iso().add(html, state, meta).render()
+  static render(html, state = {}, meta = {}, config = { markupClassName: '___iso-html___', markupElement: 'div', dataClassName: '___iso-state___', dataElement: 'div'}) {
+    return new Iso(config).add(html, state, meta).render()
   }
 
-  static bootstrap(onNode) {
+  static bootstrap(onNode, markupClassName = '___iso-html___', dataClassName = '___iso-state___') {
```
Comment (goatslacker) [inline src/iso.js]: this should probably be the config object as well

Comment (goatslacker) [inline src/iso.js]: and we can have a function that generates the default options for the config

File: examples/iso-todomvc/server.js [hunk 4]
```diff
@@ -25,7 +25,7 @@ app.get('/', function (req, res) {
 
     var markup = React.renderToStaticMarkup(React.createElement(TodoApp))
     res.render('index', {
-      html: Iso.render(markup, data)
+      html: Iso.render({}, markup, data)
```
Comment (goatslacker) [inline examples/iso-todomvc/server.js]: this needs updating

File: dist/iso.js [hunk 5]
```diff
@@ -14,32 +14,70 @@ var parse = function (node, x) {
...
+
+        return "\n" + markup + "\n" + data + "\n";
+      }
+    }
+  }, {
     render: {
       value: function render(html) {
         var state = arguments[1] === undefined ? {} : arguments[1];
         var meta = arguments[2] === undefined ? {} : arguments[2];
+        var config = arguments[3] === undefined ? { markupClassName: "___iso-html___", markupElement: "div", dataClassName: "___iso-state___", dataElement: "div" } : arguments[3];
```
Comment (goatslacker) [inline dist/iso.js]: we need to actually make sure that if someone just provides a markupClass they also get the default for the markupElement and everything else.

Comment (michaelBenin) [inline dist/iso.js]: :+1: will do

File: examples/iso-todomvc/server.js (hunk 4 above)
Comment (michaelBenin) [inline examples/iso-todomvc/server.js]: Good catch.

File: src/iso.js (hunk 3 above)
Comment (michaelBenin) [inline src/iso.js]: Should bootstrap be static? I wasn't able to access the current instance scope here. I originally had it pulling from the config. 

See: https://github.com/michaelBenin/iso/commit/621e7aa6b2b57ae80b020d8f7b8b1d34afe36ae5#diff-4e0c48295f6c143af2dfba2e4bbca24cL76

Comment (goatslacker) [inline src/iso.js]: yeah this is just sugar

otherwise you can create an Iso instance yourself, there's already a bootstrap method on the class.

**Comment by goatslacker** [comment]
I might end up switching this from class to prototypes to reduce code size.

Can you update to latest babel on this PR as well?

File: src/iso.js (hunk 3 above)
Comment (michaelBenin) [inline src/iso.js]: To be clear, we do want to remove static from this method.

Comment (goatslacker) [inline src/iso.js]: no they're two different methods, instance bootstrap does the real work where static bootstrap creates an instance and calls bootstrap.

I like the sugar of static bootstrap

**Comment by michaelBenin** [comment]
The on method should also take the config.

**Comment by michaelBenin** [comment]
@goatslacker - this should be good to go. 

Pending we're good, let me know if I should squash the commits.

**Comment by goatslacker** [comment]
lgtm

**Pull request merged**